compiler/rustc_hir_analysis/src/outlives/implicit_infer.rs

```diff
@@ -80,16 +98,18 @@ pub(super) fn infer_predicates<'tcx>(
         if !predicates_added {
             break 'outer;
         }
+        round += 1;
```

Comment:
Just for sanity, maybe check the recursion limit here against `round` (I think that could be a valid PR on its own that we could crater and land to ensure the currently indefinite-loop actually finishes